addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2688,11 +2688,15 @@ var SnippetsMenu = Widget.extend({
                     return editorToEnable;
                 }
 
+                let keepRightPanel = false;
                 if (!previewMode) {
                     this._enabledEditorHierarchy = [];
                     let current = editorToEnable;
                     while (current && current.$target) {
                         this._enabledEditorHierarchy.push(current);
+                        if (current.$target.is('img, .oe_img_bg')) {
+                            keepRightPanel = true;
```

Comment:
Instead of doing this, I did something closer to what was done in the reverted fix:
to avoid awaiting during updateUI: queue the size display update to the rpc, and do not actually wait for it